docs/architecture/deep-dives/passkeys/implementations/relying-party/prf.md

```diff
@@ -0,0 +1,101 @@
+# Passkeys for Decryption
+
+When a user logs into their Bitwarden vault, two related but distinct operations take place -
+authentication and decryption.
+
+When using a master password, the password itself serves both of these purposes - authenticating the
+user by verifying the password was entered correctly, and allowing decryption of your data through
+the derivation of an encryption key from the password. For more detail on this, see the
+[Bitwarden Security Whitepaper](https://bitwarden.com/help/bitwarden-security-white-paper/).
+
+Passkeys inherently provide the mechanism for authentication. The `prf` extension introduced in the
+WebAuthn [specification](https://w3c.github.io/webauthn/#prf-extension) provides the framework
+necessary for passkeys to be used for decryption as well.
+
+## What is PRF?
+
+At the most abstract level, account encryption and decryption requires that the user's symmetric key
+be encrypted ("protected") with another key that is available only to the client. For master
+password-based decryption, this is done with the master key, derived through a key derivation
+function from the master password.
+
+In order to use a passkey for account decryption on login, we need such a key available on the
+passkey, with which we can encrypt and decrypt the user's symmetric key. Fortunately, the `prf`
+extension to the WebAuthn spec provides exactly this functionality.
+
+The `prf` extension, which is built upon the `hmac-secret` CTAP2 extension, provides the
+specification by which credential assertion can return the output from a pseudo-random function
+(hence "PRF") that is unique to each credential on the authenticator. The Bitwarden relying party
```

Comment:
> a pseudo-random function that is unique to each credential on the authenticator

nit: it's not the function that is unique, but the secret it hashes together with the salt provided by the RP

Reply:
Just to make 100% sure - the secret itself is also unique to each credential, correct? Even without the salt?

Reply:
Yes, the secret is unique! We're using the same salt across all credentials so the only thing that differs is the secret